pc-parse/pcpatr/grammar.c

```diff
@@ -1943,13 +2024,19 @@ rulep->iNontermCount = nonterm_count;
 /*
  *  BK: check and report if A-over-A rule
  */
-if ((nonterm_count == 1) && (strcmp( lhs, rulep->apszNonterms[1] ) == 0))
+if (nonterm_count > 0 && strcmp( lhs, rulep->apszNonterms[1] ) == 0)
 	{
-	displayNumberedMessage(&sSameLeftRightCategories_m,
-			   pData->bSilent, pData->bShowWarnings, pData->pLogFP,
-			   pData->pszGrammarFilename, pData->uiLineNumber,
-			   lhs);
-	++pData->iGrammarErrorCount;
+	int obligatory = 0;
+	PATRNonterminal *nonterm;
+	for (nonterm = rhs; nonterm; nonterm = nonterm->pNext)
+		if (!nonterm->bOptional) obligatory++;
+	if (obligatory == 1) {
```

Comment:
Doesn't the single obligatory rhs term need to match the lhs for this error report?  how is this enforced?
if i'm remembering things well enough,
a -> a (b) (c) should provoke this error report
a -> (a) b (c) shouldn't provoke it if I understand things correctly (which I may not).
a -> (b) a (c) should provoke it as well, but won't pass the first screening check.